### ISSUE TITLE
Android 13: "Failed to upload" any size file on Android

### ISSUE THREAD

**Issue by moatus** [nextcloud/talk-android#2663]
### Steps to reproduce

1. Open Nextcloud Talk and choose a conversation
2. Take a picture using the in-app "Take a photo" option
3. After pushing send, nothing will appears in the chat and Android sends a "Failed to upload" notification
4. Try uploading via "Upload from device" any size file or image (testest with 40kb to 3mb) and nothing will upload (but when going the "Upload from device" route, this is no failure notification. It just don't work and nothing shows in the chat.

### Expected behaviour

Trying to attach any photo, video, or file in a conversation.

### Actual behaviour

When trying to capture or attach a photo, video, or file via Nextcloud Talk on Android, a notification appears saying "Failed to upload": 
![image](https://user-images.githubusercontent.com/15126267/210027049-560edb0d-88ce-4c66-a58d-7e9a6569367c.png)

It doesn't matter if you use "Take photo" or "Upload from device", none of the attachment options seems to work (though the failure notification only show up with "Take photo" and "Take video". Share contact doesn't work either. The only attachment that works is "Share location".

I've tested with a Pixel 4a and a Samsung Galaxy A52 and neither work. Both phones and the server are all on the same local network.

### Device brand and model

Pixel 4a

### Android version

13

### Nextcloud Talk app version

15.0.3

### Nextcloud server version

25.0.2

### Talk version

15.0.2

### Custom Signaling server configured

None

### Custom TURN server configured

None

### Custom STUN server configured

None

**Comment by moatus** [comment]
Is this issue because Talk hasn't been updated for Android 13? Just saw this on the product page: https://github.com/orgs/nextcloud/projects/57/views/11

**Comment by mahibi** [comment]
HI @moatus thanks for the report!
Just checked and indeed there seem to be problems while uploading with android 13! We will increase the importance to upgrade!

I didn't yet had a closer look but most likely there are follow-up bugs caused by the new granular-media-permissions
https://developer.android.com/about/versions/13/behavior-changes-13


i got this when trying to take a photo:
```
D/ImageCapture: Send image capture request [current, pending] = [0, 1]
D/ImageCapture: issueTakePicture
D/Camera2CameraImpl: {Camera@187ff93[id=0]} Issue capture request
D/CaptureSession: Issuing capture request.
D/DeferrableSurface: use count-1,  useCount=0 closed=false androidx.camera.core.SurfaceRequest$2@f2cea1c
D/DeferrableSurface: Surface no longer in use[total_surfaces=2, used_surfaces=1](androidx.camera.core.SurfaceRequest$2@f2cea1c}
D/DeferrableSurface: use count-1,  useCount=0 closed=false androidx.camera.core.impl.ImmediateSurface@361cef0
D/DeferrableSurface: Surface no longer in use[total_surfaces=2, used_surfaces=0](androidx.camera.core.impl.ImmediateSurface@361cef0}
D/Camera2CameraImpl: CameraDevice.onError(): 0 failed with ERROR_CAMERA_DEVICE while in OPENED state. Will attempt recovering from error.
D/Camera2CameraImpl: Attempt to reopen camera[0] after error[ERROR_CAMERA_DEVICE]
D/Camera2CameraImpl: {Camera@187ff93[id=0]} Transitioning camera internal state: OPENED --> REOPENING
D/CameraStateRegistry: Recalculating open cameras:
    Camera                                       State                 
    -------------------------------------------------------------------
    Camera@187ff93[id=0]                         OPENING               
    Camera@fa901ef[id=1]                         UNKNOWN               
    -------------------------------------------------------------------
    Open count: 1 (Max allowed: 1)
D/CameraStateMachine: New public camera state CameraState{type=OPENING, error=StateError{code=3, cause=null}} from OPENING and StateError{code=3, cause=null}
D/CameraStateMachine: Publishing new public camera state CameraState{type=OPENING, error=StateError{code=3, cause=null}}
D/Camera2CameraImpl: {Camera@187ff93[id=0]} Resetting Capture Session
E/CameraCaptureSession: Session 0: Exception while stopping repeating: 
    android.hardware.camera2.CameraAccessException: CAMERA_ERROR (3): The camera device has encountered a serious error
        at android.hardware.camera2.impl.CameraDeviceImpl.checkIfCameraClosedOrInError(CameraDeviceImpl.java:2429)
        at android.hardware.camera2.impl.CameraDeviceImpl.stopRepeating(CameraDeviceImpl.java:1329)
        at android.hardware.camera2.impl.CameraCaptureSessionImpl.close(CameraCaptureSessionImpl.java:579)
        at androidx.camera.camera2.internal.SynchronizedCaptureSessionBaseImpl.close(SynchronizedCaptureSessionBaseImpl.java:474)
        at androidx.camera.camera2.internal.CaptureSession.release(CaptureSession.java:420)
        at androidx.camera.camera2.internal.Camera2CameraImpl.releaseSession(Camera2CameraImpl.java:524)
        at androidx.camera.camera2.internal.Camera2CameraImpl.resetCaptureSession(Camera2CameraImpl.java:1209)
        at androidx.camera.camera2.internal.Camera2CameraImpl.closeCamera(Camera2CameraImpl.java:438)
        at androidx.camera.camera2.internal.Camera2CameraImpl$StateCallback.reopenCameraAfterError(Camera2CameraImpl.java:1678)
        at androidx.camera.camera2.internal.Camera2CameraImpl$StateCallback.handleErrorOnOpen(Camera2CameraImpl.java:1630)
        at androidx.camera.camera2.internal.Camera2CameraImpl$StateCallback.onError(Camera2CameraImpl.java:1606)
        at androidx.camera.camera2.internal.CameraDeviceStateCallbacks$ComboDeviceStateCallback.onError(CameraDeviceStateCallbacks.java:121)
        at android.hardware.camera2.impl.CameraDeviceImpl.notifyError(CameraDeviceImpl.java:1809)
        at android.hardware.camera2.impl.CameraDeviceImpl.$r8$lambda$KBQCqQRdhVVn7uHI9Xdha6OqnsU(Unknown Source:0)
        at android.hardware.camera2.impl.CameraDeviceImpl$$ExternalSyntheticLambda0.accept(Unknown Source:8)
        at com.android.internal.util.function.pooled.PooledLambdaImpl.doInvoke(PooledLambdaImpl.java:281)
        at com.android.internal.util.function.pooled.PooledLambdaImpl.invoke(PooledLambdaImpl.java:204)
        at com.android.internal.util.function.pooled.OmniFunction.run(OmniFunction.java:97)
        at androidx.camera.core.impl.utils.executor.SequentialExecutor$1.run(SequentialExecutor.java:111)
        at androidx.camera.core.impl.utils.executor.SequentialExecutor$QueueWorker.workOnQueue(SequentialExecutor.java:231)
        at androidx.camera.core.impl.utils.executor.SequentialExecutor$QueueWorker.run(SequentialExecutor.java:173)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
        at java.lang.Thread.run(Thread.java:1012)
D/Camera2CameraImpl: {Camera@187ff93[id=0]} Releasing session in state REOPENING
D/Camera2CameraImpl: {Camera@187ff93[id=0]} CameraDevice.onDisconnected()
```

**Comment by maxcco** [comment]
I cannot reproduce on Samsung s10e running LOS 20 (android 13)
Every upload work well from same network or different

**Comment by lbdroid** [comment]
Also unable to reproduce, Pixel 6pro and 7pro, GrapheneOS. Further details required before concluding that this is related to Android version.

**Comment by moatus** [comment]
Tested it again @lbdroid @maxcco and it is uploading now. Server hasn't been restarted and conditions are the same, media is now uploading.

If it shows up gain, I'll update. Thanks for looking into it!

**Comment by lbdroid** [comment]
@moatus ; are you certain that the talk android version wasn't 15.0.0 or 15.0.1 awhen you were having the issue and subsequently uodated? There was an upload issue that affected files over 1MB that was fixed in version 15.0.2.

**Comment by moatus** [comment]
@lbdroid I am curtain it was 15.0.2. I had read about the issue with less than 1MB files, which is why I tested with files from 40kb to 3mb. Any file I tried to send I would get the error message I posted above.

**Comment by ondrejlohnisky** [comment]
I am having the same problem on v16.0.1 
and i have this info related to this issue from nginx error log:
`2023/05/02 23:44:07 [info] 1703#1703: *69 client canceled stream 7 while sending request to upstream, client: myIP server: myhost.com, request: "GET /ocs/v2.php/apps/spreed/api/v1/chat/badc453h?lastKnownMessageId=1561&includeLastKnown=0&limit=100&setReadMarker=0&timeout=30&lookIntoFuture=1 HTTP/2.0", upstream: "fastcgi://unix:/var/run/php/php8.1-fpm.sock:", host: "myhost.com`

**Comment by mahibi** [comment]
> I am having the same problem on v16.0.1 and i have this info related to this issue from nginx error log: `2023/05/02 23:44:07 [info] 1703#1703: *69 client canceled stream 7 while sending request to upstream, client: myIP server: myhost.com, request: "GET /ocs/v2.php/apps/spreed/api/v1/chat/badc453h?lastKnownMessageId=1561&includeLastKnown=0&limit=100&setReadMarker=0&timeout=30&lookIntoFuture=1 HTTP/2.0", upstream: "fastcgi://unix:/var/run/php/php8.1-fpm.sock:", host: "myhost.com`

This request is about getting chat messages, so it's not directly related to the upload itself.
Do you see the error notification? So you also don't see the upload when leaving the conversation and enter again right?

So if it's related to nginx, maybe https://docs.nextcloud.com/server/latest/admin_manual/installation/nginx.html could contain some hints for configuration

**Comment by ondrejlohnisky** [comment]
Nextcloud logging does not register this error. The only information about this is in notification as "Upload failed" and this info log in error logs of nginx. But I also read somewhere that Nextcloid Talk is not 100% supported on android 13. So i am now waiting on 4th week of May for releass.

**Comment by surfrock66** [comment]
I know this is old but still open, I am experiencing this too.

Nextcloud 17.1.2 on Ubuntu Server 22.04
Textcloud Talk version 17.1.1

Android 12 on OnePlus 7 Pro
Nextcloud Talk Android App 17.1.0 installed from FDroid

Same thing, same notification error, nothing whatsoever in the server logs.  The file I'm trying to upload is a 163kb jpg.

**Comment by surfrock66** [comment]
That path is strange; each user has a "Talk" folder visible under files.  My user is the 6c7 user; I can browse to my nextcloud url at /remote.php/dav/ and get "This is the WebDAV interface. It can only be accessed by WebDAV clients such as the Nextcloud desktop sync client."  

I am set up with LDAP, and I'm noticing that the username and the display name are different; could that matter?

![image](https://github.com/nextcloud/talk-android/assets/1345467/d4cbe15e-1cdd-49cf-8fc3-da949224fc85)

**Comment by SlimTux** [comment]
Why has this issue been closed? The problem has not been solved yet!

**Comment by SlimTux** [comment]
> Also unable to reproduce, Pixel 6pro and 7pro, GrapheneOS. Further details required before concluding that this is related to Android version.

I'm having the exact same issue with the newest app version

**Comment by madsmki** [comment]
I have exact same issue.

Android 12
Talk App v18.1.0
Nextcloud server 28.0.2
PHP 8.2

I am new to this forum. Should I file a new issue or will this one be reopened?

**Comment by mahibi** [comment]
Thanks for reporting.
Is it possible to upload files via web for the same user?
Does it still fail afterwards on android?

**Comment by madsmki** [comment]
I have no problems with uploading from web. Android still fails, but actually its possible to upload small files less than 1 MB, so it could be related to the issue that "was fixed" in version 15.0.2.

**Comment by mahibi** [comment]
Does it still happen with app version 19.0 ?

**Comment by Stratosfear1** [comment]
Got the same issue and sure, it also occurs with app version 19.0.1 on different Samsung devices with Android 12, 13 and 14, Server Nextcloud 29.0.1.
But not always. Sometimes the behaviour is different. But don't know the circumstances when it will work and when not. Possible it depends on network used. Sometime it helps, switching from wlan to mobile data or vice versa. But some wlan will work, some not. Sometimes mobile data will work, sometimes not.

**Comment by xcpky** [comment]
the exact same issue happens here on android 14

**Comment by Adefey** [comment]
Hi
I have the exact same issue. Same 403 status in response. After selecting a big file (picture from camera for example 3..7 MB), it uploads but then I get "Failed to upload %filename" notifiation. The picture itself is successfully loaded to Talk folder of my user

**Comment by mahibi** [comment]
> Hi I have the exact same issue. Same 403 status in response. After selecting a big file (picture from camera for example 3..7 MB), it uploads but then I get "Failed to upload %filename" notifiation. The picture itself is successfully loaded to Talk folder of my user

moved the issue that you created to server: https://github.com/nextcloud/server/issues/59108

I plan to have a closer look on file uploads when https://github.com/nextcloud/talk-android/pull/5635 is done (incl upload progress in chat itself). Though if it's server side issues i can just improve the error handling..

**Comment by mahibi** [comment]
The uploading issues should be fixed at latest with v25.0. (out this week).
If there are still issues after updating, please open new github tickets
